tests/phpunit/integration/Core/Admin/DashboardTest.php

```diff
@@ -70,4 +70,216 @@ function ( $cap ) {
 		$this->assertTrue( wp_script_is( 'googlesitekit-wp-dashboard', 'enqueued' ) );
 		$this->assertTrue( wp_style_is( 'googlesitekit-wp-dashboard-css', 'enqueued' ) );
 	}
+
+	/**
+	 * @group dashboard_widget
+	 */
+	public function test_render_googlesitekit_wp_dashboard_loading_container() {
+		$dashboard = new Dashboard( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+		$method    = new \ReflectionMethod( $dashboard, 'render_googlesitekit_wp_dashboard' );
+		$method->setAccessible( true );
+
+		ob_start();
+		$method->invoke( $dashboard );
+		$output = ob_get_clean();
+
+		$dom = new \DOMDocument();
+		$dom->loadHTML( $output );
+
+		$this->assertNotEmpty( $dom->getElementById( 'js-googlesitekit-wp-dashboard' ) );
+
+		$xpath = new \DOMXPath( $dom );
+		$query = '//div[contains(concat(" ", normalize-space(@class), " "), " googlesitekit-wp-dashboard ") and contains(concat(" ", normalize-space(@class), " "), " googlesitekit-wp-dashboard-loading ")]';
+
+		$elements = $xpath->query( $query );
+
+		$this->assertEquals( 1, $elements->length );
+	}
+
+	/**
+	 * @group dashboard_widget
+	 */
+	public function test_render_googlesitekit_wp_dashboard_only_search_console_connected() {
+		// Only set the search console module as connected.
+		$active_modules = array(
+			'search-console' => $this->getMockBuilder( 'Google\Site_Kit\Core\Modules' )
+				->disableOriginalConstructor()
+				->setMethods( array( 'get_active_modules' ) )
+				->getMock(),
+		);
+
+		$active_modules['search-console']
+			->method( 'get_active_modules' )
+			->willReturn(
+				array(
+					'search-console' => (object) array(
+						'is_connected' => function () {
+							return true;
+						},
+					),
+				)
+			);
+
+		$admin_id = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $admin_id );
+		$dashboard = new Dashboard( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+		$method    = new \ReflectionMethod( $dashboard, 'render_googlesitekit_wp_dashboard' );
+		$method->setAccessible( true );
+
+		$mockAuthentication = $this->getMockBuilder( \stdClass::class )
+			->addMethods( array( 'is_authenticated' ) )
+			->getMock();
+		$mockAuthentication->method( 'is_authenticated' )->willReturn( true );
```

Comment:
In addition to the point raised in my comment on the test above, it would be preferable to avoid mocking these methods. We do prefer a more integrated approach to testing, only mocking where it's necessary. I think these tests can be rewritten without the mocks. Please take a look.